### DISCUSSION TITLE
Can buttons be added to the status bar?

### DISCUSSION THREAD

**Discussion by reduckted** [VsixCommunity/Community.VisualStudio.Toolkit#243] [Q&A] [answered]
When you have a solution open that is in a Git repository, the status bar shows information about the repository:

![image](https://user-images.githubusercontent.com/10321525/145764986-92bb84a2-653b-4480-ad1e-c15b25a82e72.png)

Is there a public API to do that? The `IVsStatusbar` doesn't let you do that, and I'm curious if there's a different service that needs to be used. Perhaps a MEF export?

In the past, I've added custom icons/buttons to the status bar by finding the status bar control in the WPF visual tree and manually inserting child elements, but that's very hacky and I'm hoping for a better way. 😁

**Comment by madskristensen** [comment, marked as answer]
There is no public API for this. In fact, there is no internal API for it either. Going through the visual tree is the only way to do it
